### PR TITLE
[Docs] Clad Tutorial

### PR DESCRIPTION
**Pull request by Ris-Bali**
This PR aims to complete clad's tutorial documentation.

### CODE REVIEW

**Comment by Ris-Bali** [comment]
The error is now fixed

File: docs/userDocs/source/user/UsingClad.rst [hunk 1]
```diff
@@ -217,9 +217,23 @@ Few important things to note about ``clad::hessian``:
...
 
-   Add details for computing hessian of array ranges.
+.. code-block:: cpp
+  
+  #include "clad/Differentiator/Differentiator.h"
+  double fn(double x, double arr[2]) {
+     return x * arr[0] * arr[1];
+  }
+  int main() {
+  auto fn_hessian = clad::hessian(fn, "x, arr[0:1]");
```
Comment (parth-07) [inline docs/userDocs/source/user/UsingClad.rst:230]: Please format this code.

File: docs/userDocs/source/user/tutorials.rst [hunk 2]
```diff
@@ -1,6 +1,168 @@
...
+   #include "clad/Differentiator/Differentiator.h"
+   #include <iostream>
+
+
+   double func(int x) {
+   return x * x;
+   }
+
+   int main() {
+   /*Calling clad::differentiate to get the forward mode derivative of
```
Comment (parth-07) [inline docs/userDocs/source/user/tutorials.rst:24]: Can you please format the code everywhere using clang-format?

File: docs/userDocs/source/user/tutorials.rst [hunk 3]
```diff
@@ -1,6 +1,168 @@
...
+   /*Calling clad::differentiate to get the forward mode derivative of 
+   the given mathematical function*/
+   auto d_func = clad::differentiate(func, "x");
+   // execute the generated derivative function.
+   std::cout << d_func.execute(/*x =*/3) <<std::endl;
+   // Dump the generated derivative code to std output.
+   d_func.dump();
+   }
+
+Here we are differentiating a function `func` which takes an input x and
```
Comment (parth-07) [inline docs/userDocs/source/user/tutorials.rst:33]: `x` should be in code-formatting here.

File: docs/userDocs/source/user/tutorials.rst [hunk 4]
```diff
@@ -1,6 +1,168 @@
...
+In the above example we are differentiating w.r.t `x and y` we can also 
+differentiate w.r.t to single argument i.e. either `x` or `y` as `clad::gradient(f, "x")` 
+not writing any argument i.e. `clad::gradient(f)` will result in differentiation 
+of the function w.r.t to each input. 
+
+
+**The Hessian Mode**
+
+Clad can also produce an hessian matrix through the `clad::hessian` API call.
+This API call is more or less similar to the reverse mode API call and differs
```
Comment (parth-07) [inline docs/userDocs/source/user/tutorials.rst:68]: Hessian mode is not more or less similar to the reverse mode API call.

**Review by parth-07 — CHANGES_REQUESTED** [review]

**Review by parth-07 — APPROVED** [review]
Please squash the commits into one commit and improve the commit message.

**Pull request merged**